experiments/AMIP/components/atmosphere/climaatmos.jl

```diff
@@ -21,6 +21,18 @@ struct ClimaAtmosSimulation{P, Y, D, I} <: Interfacer.AtmosModelSimulation
 end
 Interfacer.name(::ClimaAtmosSimulation) = "ClimaAtmosSimulation"
 
+hasradiation = function (integrator)
+    return !isnothing(integrator.p.atmos.radiation_mode)
+end
+
+hasmoisture = function (integrator)
+    return !(integrator.p.atmos.moisture_model isa CA.DryModel)
+end
+
+zero_scalar_half(integrator) = zeros(axes(CC.Fields.level(integrator.p.scratch.ᶠtemp_scalar, CC.Utilities.half)))
```

Comment:
Also, these functions will allocate since they're calling `zeros`. We should avoid this since they're called in `get_field`. Maybe we could have temp fields on each of these spaces and use them here?